modules/sdk/src/main/scala/org/tessellation/sdk/infrastructure/consensus/ConsensusStateUpdater.scala

```diff
@@ -241,7 +243,7 @@ object ConsensusStateUpdater {
 
           maybeSignedArtifact.traverse { signedArtifact =>
             val newState = state.copy(status = Finished(signedArtifact))
-            consensusFns.consumeSignedMajorityArtifact(signedArtifact).map { _ =>
+            consensusFns.consumeSignedMajorityArtifact(signedArtifact).flatTap(_ => updateCluster()).map { _ =>
```

Comment:
Let's decouple collateral from consensus. Please subscribe on the stream of new heads in the `GlobalSnapshotStorage` by storing the head in the `fs2.concurrent.SignallingRef`. This way you hook up to only a single place where the snapshots are stored.

Reply:
I have added CollateralDaemon to accomplish this suggestion. Thanks!